kusto/kcsb.go

```diff
@@ -423,6 +431,28 @@ func (kcsb *ConnectionStringBuilder) getTokenProvider() (*TokenProvider, error)
 
 }
 
+func (kcsb *ConnectionStringBuilder) initialiseWithCloudInfo(ci CloudInfo) {
```

Comment:
this feels wrong and too broad.
CloudInfo should not autoComplete KCSB, it should only affect the construction orf the specific credential and since those don't share all below decisions it's error prone.

Why not simply fill in the case-specific values in the token provider configuration object per each provider created
?